pkg/networkmanager/ip-settings.jsx

```diff
@@ -70,6 +70,15 @@ export const IpSettingsDialog = ({ topic, connection, dev, settings }) => {
     const [method, setMethod] = useState(params.method);
     const [routes, setRoutes] = useState(params.routes ? params.routes.map(addr => ({ address: addr[0], netmask: addr[1], gateway: addr[2], metric: addr[3] })) : []);
 
+    // Some ipv4/ipv6 method values are not suported for WireGuard,
+    // so remove them from the (ipv4/ipv6)_method_choices variables above
+    if (dev.DeviceType === 'wireguard') {
+        const unsupported_ipv4_method_values = ['auto', 'link-local', 'shared'];
+        const unsupported_ipv6_method_values = ['auto', 'dhcp', 'shared'];
+        ipv4_method_choices = ipv4_method_choices.filter(item => !unsupported_ipv4_method_values.includes(item.choice));
+        ipv6_method_choices = ipv6_method_choices.filter(item => !unsupported_ipv6_method_values.includes(item.choice));
+    }
```

Comment:
Changing constants on runtime which are exported feels very brittle.

Reply:
Can these be turned into a `function ipv4_method_choices(device_type)`?

Reply:
Using a function indeed sounds better, thanks! I see that the IPv4 methods are a subset of the IPv6 methods:
```js
export let ipv4_method_choices =
    [
        { choice: 'auto', title: _("Automatic (DHCP)") },
        { choice: 'link-local', title: _("Link local") },
        { choice: 'manual', title: _("Manual") },
        { choice: 'shared', title: _("Shared") },
        { choice: 'disabled', title: _("Disabled") }
    ];

export let ipv6_method_choices =
    [
        { choice: 'auto', title: _("Automatic") },
        { choice: 'dhcp', title: _("Automatic (DHCP only)") },
        { choice: 'link-local', title: _("Link local") },
        { choice: 'manual', title: _("Manual") },
        { choice: 'ignore', title: _("Ignore") },
        { choice: 'shared', title: _("Shared") },
        { choice: 'disabled', title: _("Disabled") }
    ];
```

Only difference is "auto" translates to "Automatic (DHCP)" is IPv4 and to just "Automatic" in IPv6. The value "auto" in both IPv4/6 seems to be exactly the same. From NM doc:
>NMSettingIP4Config and NMSettingIP6Config both support "disabled", "auto", "manual", and "link-local"...

If we use the same text for "auto" in both IPv4/IPv6, we can also merge the list into one while we're at it...

Reply:
Sorry, I think that was bad advice. "Auto" for IPv6 is often *not* DHCP, but SLAAC, or possibly some other methods. That's why it has a separate "dhcp" mode. So we may be able to do it the other way around and name both "Automatic", but when in doubt, let's just leave them as-is?

Reply:
Thanks for the feedback @martinpitt! I did some research into this and it indeed looks like "auto" in IPv6 doesn't mean DHCP. TIL.

In case of IPv6 it says [here](https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/blob/main/src/libnm-core-impl/nm-setting-ip6-config.c?ref_type=heads#L28-29):
> If "auto" is specified then the appropriate automatic method (PPP, router advertisement, etc) is used for the device

Similar mentioned [here](https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/blob/main/src/libnm-core-impl/nm-setting-ip6-config.c?ref_type=heads#L1413) as well, which looks like part of a documentation, but I can't find the rendered version.

I also looked into IPv4, and it says:
> Enables automatic IPv4 address assignment from DHCP, PPP, or similar services.

Source: https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/blob/main/src/libnm-core-impl/nm-setting-ip4-config.c?ref_type=heads#L1285

So, IPv4 "auto" is more than just DHCP it seems, then your second suggestion of dropping (DHCP) from IPv4, might be the right way to go.

WDYT?

Reply:
Works for me -- cross check, @mvollmer ?

Reply:
I think "DHCP" is a very familiar term and will reassure people that they are getting what they think they are getting. But if it is only correct for certain types of interfaces, I don't mind dropping it.